pyproject.toml

```diff
@@ -89,12 +86,10 @@ include = ["snakemake", "snakemake.*"]
   "*.rst",
 ]
 
-[tool.versioneer]
-VCS = "git"
-style = "pep440"
-tag_prefix = "v"
-versionfile_build = "snakemake/_version.py"
-versionfile_source = "snakemake/_version.py"
+[tool.setuptools_scm]
+version_file = "src/snakemake/_version.py"
+# Ignore Git hash
+local_scheme = "no-local-version"
```

Comment:
`8.30.1.dev32` is PEP 440 compliant, while `8.30.1.dev32+gd6d6a9d7.d20250312` is not.

Reply:
Hm. Maybe I'm wrong on that given `pip` doesn't error on it

```python
>>> from pip._vendor.packaging.version import Version
>>> Version("8.30.1.dev32")
<Version('8.30.1.dev32')>
>>> Version("8.30.1.dev32+gd6d6a9d7.d20250312")
<Version('8.30.1.dev32+gd6d6a9d7.d20250312')>
```

but commit hashes should not ever enter PyPI names.